### ISSUE TITLE
Cosmetic suggestions

### ISSUE THREAD

**Issue by Khagrin2** [NiteMoves/tcec-web-public#6]
I think many viewers would like to see evals, clocks and "whose turn" info at a glance.  I would suggest emphasizing those elements.  Perhaps a "check" alongside engine logo's to clearly indicate which engine's turn it is.  Perhaps evals, clocks and engine names can have different color/font or somesuch to make them stand out a bit.  I would separate the "main clock" from the "move time" clocks a bit.   Thanks for providing this means of making suggestions!

**Comment by z1bnc** [comment]
I also agree with these suggestions. It looks like the side to move has a black color font for the clocks and the other side has a slightly lighter grey ?

**Comment by gitlost** [comment]
Could you do a mock-up of some suggestions @Khagrin2 (like you did for the main layout redesign https://imgur.com/a/jC2sBKB)?

**Comment by Khagrin2** [comment]
I've been really busy ... hardly time to even watch the games for the past
couple of days!  That said, I can sure try to do that sometime reasonably
soon.

**Comment by Khagrin2** [comment]
Here is a mockup of the center that might give you some ideas of what I was thinking.  I tried to keep the suggested changes minimal to make an implementation of such tweaks as simple as possible to achieve the desired effects.  Essentially I moved the evals, added arrows to indicate whose move it currently is, and positioned the clocks a bit differently.  

![image](https://user-images.githubusercontent.com/40367843/41754910-96e6977a-75a2-11e8-8a2f-55cf576ca24b.png)

**Comment by gitlost** [comment]
Very nice @Khagrin2, will give that a shot, but probably be a week or 2 (or 3), ta!

**Comment by Khagrin2** [comment]
This could as well be in the other "Cosmetics + Material Diff" issue.  I recalled a suggestion I made for YLCET regarding "move indicator".  It was never implemented, but here it might provide an elegant and useful indicator that I will call a "time bar".  It amounts to placing three z-ordered bars under each clock display.  A bottom layer and indicates "total time for the game", no need for calculating this ... it is just a the width of the logo images (background color if not moving, light gray if on-the-move).  The second bar (red or perhaps blue when the engine is on-the-move, and mid-gray if not) is next in the z-order and indicates "time used so far".  The last bar (white if the engine is on-the-move, gray if not) indicates "time used for current move".  For the thinking engine, the white bar is updated each second (as per elapsed time clock), so it tends to grow longer as time passes.  Here is the mockup:

![image](https://user-images.githubusercontent.com/40367843/41871259-655b5ef6-788c-11e8-842b-14f877717585.png)

Some benefits as game progresses:

- The viewer easily sees which computer is on-the-move.
- The viewer can easily see the relative "time used vs. time remaining".
- The viewer can see the relative "time for this move vs. time used so far'.
- The viewer can, at a glance, see which engine has used more time and which engine is on-the-move.

So ... how to determine the "length" of the bars (assuming all are at the same X-offset and z-ordered properly)?  Here are some example calculations I think would work:

The width of the engine logo (and hence "game time" bar) = 100px.
Note that at the start of a 120 minute (7200 second) game, the pixels/seconds ratio would be (100 / 7200) = ~0.014 ... i.e. depending on how your round things.  ~70 seconds of time amounts to 1 pixel of "bar".  But ... this ratio will change during the course of the game due to the 15 second increment per move.  

At any given second during the game, the length of the "time used" and "current move time" bars can be calculated for one of the engines as shown in this example: 

   PX = 100   ... this is the width of each container that holds the engine/clock info

   N_Moves = Ex: 30    ... number of moves engine has previously made
   PMI = 15 seconds ... the per move increment
   T_TotalIncrementBonusEarned = (N_Moves * PMI) = 450 seconds

   T_PrevMoves = Total seconds engine has used on previous moves (Ex:  60 minutes = 3600 seconds)
   T_Curmove = Total seconds engine has used on current move (Ex: 4 minutes = 240 seconds)
   T_TotalUsedByEngine = (T_PrevMoves + T_Curmove) = (3600 + 240) = 3840

   T_Game = 7200  ... seconds for a 120 minute game
   T_GamePlusIncrementsEarned = (T_Game + T_TotalIncrementBonusEarned) = (7200 + 450) = 7650

   Pixels_Per_Second_Ratio = (PX / T_GamePlusIncrementsEarned) = (100 / 7650) = ~0.013

"Used" bar width =  T_PrevMoves * Pixels_Per_Second_Ratio = (3600 * 0.013) = ~47px
"Current" bar width = T_TotalUsedByEngine * Pixels_Per_Second_Ratio (3840 * 0.013) = ~50px

**Comment by gitlost** [comment]
Good stuff @Khagrin2 , was thinking also of maybe having mini vertical eval bars along and beside each engine logo? Then moving the eval field above/below them? This would make the column a bit less crowded. Eg (not sure about eval field format):

![khagrin2-clock](https://user-images.githubusercontent.com/481982/41922552-44c055da-795d-11e8-9701-30b88875c1a2.png)

(To be honest I don't get "Material Diff" graveyards but it seems there's a big demand for them so "must have" I suppose (was thinking of just having them in a tooltip on the current move, along with a pretty move list)...)

**Comment by Khagrin2** [comment]
Very nice ... the little eval bars are tasty!

**Comment by Nordlandia** [comment]
What's your opinion ICC app material indicator?

![minus1forwhite](https://user-images.githubusercontent.com/26534454/41974414-32496fd4-7a18-11e8-9309-4e11095a8b1f.png)
![plus1forblack](https://user-images.githubusercontent.com/26534454/41974431-3b435064-7a18-11e8-8273-c06b1165d557.png)



![suggestion](https://user-images.githubusercontent.com/26534454/41974392-218280c8-7a18-11e8-8f47-17b1dc6d684c.png)

**Comment by Khagrin2** [comment]
I, like another opinion previously expressed, have never really seen the
need for any kind of "material diff" indication or display.  But,I
understand that some folks do like such so I'm not opposed to a tasteful
indication of some kind.  That said, I don't really like the "guage" much.
I think it clutters your nice clean GUI, and doesn't really add any
information for the viewer.

**Comment by Nordlandia** [comment]
Displaying the example imbalance:  NN  RPP +1 is more than adequate. Maybe the indicator is unnecessary after all.

**Comment by NiteMoves** [comment]
Maybe we need to add @Khagrin2 to the private repo so he can do PRs. :)
